runtime/src/accounts_db.rs

```diff
@@ -419,6 +419,10 @@ pub struct AccountsDB {
     /// Keeps tracks of index into AppendVec on a per slot basis
     pub accounts_index: AccountsIndex<AccountInfo>,
 
+    pub uncleaned_keys: Mutex<HashMap<Slot, Vec<Pubkey>>>,
```

Comment:
I think we can also remove dead/unrooted slots from `uncleaned_keys` in `purge_slots()` as part of slot cleanup when slots are dropped via `Bank::drop()`,